app/controllers/sites/new.js

```diff
@@ -0,0 +1,68 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { combineFullAddress } from 'frontend-contactgegevens-loket/models/address';
+import { setEmptyStringsToNull } from 'frontend-contactgegevens-loket/utils/empty-string-to-null';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class CreateSitesNewController extends Controller {
+  @service router;
+  @service store;
+  @tracked isPrimarySite = false;
+  @action
+  async createSite(event) {
+    event.preventDefault();
+    const { address, primaryContact, secondaryContact, site, adminUnit } =
+      this.model;
+    address.fullAddress = combineFullAddress(address);
+    await primaryContact.save();
+    await secondaryContact.save();
+    await address.save();
+    site.contacts = [primaryContact, secondaryContact];
+    site.address = address;
+    console.log('Primary site', this.isPrimarySite);
+    await site.save();
+    let nonPrimarySites = await adminUnit.sites;
+
+    if (this.isPrimarySite) {
+      let previousPrimarySite = await adminUnit.primarySite;
+
+      if (previousPrimarySite) {
+        nonPrimarySites = [previousPrimarySite];
```

Comment:
I don't think this is working, because you should add the previousPrimarySite to the array of secondary sites, this will only work if there's one primary and one secondary site